src/api-service/__app__/onefuzzlib/azure/vm.py

```diff
@@ -181,16 +187,20 @@ def has_components(name: str) -> bool:
     return False
 
 
-def delete_vm_components(name: str) -> bool:
+def delete_vm_components(name: str, nsg: Optional[NSG]) -> bool:
```

Comment:
I'm not sure it makes sense to pass int he entire NSG. Is there a way to do this where the VMs location is used to retrieve the NSG for that region?

Reply:
VM's NIC is associated with an NSG. From within VM code VM does not know which NSG it is associated with. 
In case there are multiple NSGs for a region - VM needs to know which one it needs to dissociate from.

Unless we have explicit agreement that there is never more than one NSG per region

Reply:
Okay yeah I suppose we'd have to say that there's only one NSG per pregion.